### DISCUSSION TITLE
[json.exception.type_error.305] cannot use operator[] with a string argument with string

### DISCUSSION THREAD

**Discussion by xo1337** [nlohmann/json#3152] [Q&A]
Hi there. I've been trying to parse a JSON object from a C# websocket server. I've sent data to my client via `NewtonSoft.Json` by doing `JsonConvert.SerializeObject(new List<string, string>({"status, "success"}));`

And i try to parse my object from my server and get this exception: `[json.exception.type_error.305] cannot use operator[] with a string argument with string`. I am not understanding why. The JSON object is VALID but when i call the nlohmann::json::get function with any data type it throws this exception.

Heres the server json object:
```json
"{\n  \"status\": \"succes\",\n  \"id\": 1}"
```

If anyone can help, that would be awesome. :)

**Comment by nlohmann** [comment]
Can you share the C++ code you are calling that triggers the exception?

**Reply by xo1337** [reply]
Sure

```cpp
auto response = json::parse ( Request::Get ( "localhost/testing/test.php?action=testing" ) );

		if ( response["status"].get<std::string> ( ) == "success" )
		{
			try
			{
				auto object = json::parse ( response["internal_json_object"].get<json> ( ).dump ( ) ); // auto object = response["internal_json_object"].get<json> ( ); // doesn't work either
				std::cout << "Object name: " << object["name"].get<std::string> ( ) << std::endl;
			}
			catch ( json::exception& e )
			{
				printf_s ( "Exception JSON: %s\n" , e.what ( ) );
			}
		}
```

**Reply by xo1337** [reply]
Basically in my localhost JSON response i have a json object inside of a JSON object.

**Reply by nlohmann** [reply]
What is the value of `response`?

**Reply by xo1337** [reply]
> ```json
> "{\n  \"status\": \"succes\",\n  \"id\": 1, \"name\":\"test\"}"
> ```

**Reply by xo1337** [reply]
I don't really understand why it has all the \", i've tried removing the \" and just putting " and removing the \n but i still get the exception thrown.

**Reply by nlohmann** [reply]
What do you expect for `response["internal_json_object"]` to happen if `response` does not have key `internal_json_object`?

**Reply by xo1337** [reply]
It does hava a key. `"{\n  \"status\": \"succes\",\n  \"id\": 1, \"name\":\"test\"}"`

**Reply by nlohmann** [reply]
Where is the key `internal_json_object` in `"{\n \"status\": \"succes\",\n \"id\": 1, \"name\":\"test\"}"`?

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?
